csrc/fusion_profiler.cpp

```diff
@@ -378,6 +379,61 @@ uint32_t SegmentProfiler::segmentId() const {
   return segment_id_;
 }
 
+auto FusionProfile::toTuple(const FusionProfile& prof, size_t seg_id) {
+  NVF_CHECK(
+      !prof.kernel_profiles.empty(),
+      "Cannot convert FusionProfile to a tuple containing CUPTI gathered stats!");
+  NVF_CHECK(
+      seg_id < prof.kernel_profiles.size(),
+      "Invalid seg_id for FusionProfile. Segments: ",
+      prof.kernel_profiles.size(),
+      " seg_id: ",
+      seg_id);
+  auto& kp = prof.kernel_profiles[seg_id];
```

Comment:
Is `const auto&` safe here? It changes the return type but I think the tuple type is never explicitly stated anyway. If it doesn't work, please ignore.